src/reconcile/rec_col.c

```diff
@@ -1419,19 +1419,19 @@ __wt_rec_col_var(
                     data = upd->data;
                     size = upd->size;
                     /*
-                     * When a mixed mode tombstone is getting written to disk, remove any historical
-                     * versions that are greater in the history store for this key.
+                     * When a tombstone without a timestamp is written to disk, remove any
+                     * historical versions that are greater in the history store for this key.
                      */
-                    if (upd_select.mm_tombstone && r->hs_clear_on_tombstone)
+                    if (upd_select.no_ts_tombstone && r->hs_clear_on_tombstone)
                         WT_ERR(__wt_rec_hs_clear_on_tombstone(session, r, src_recno, NULL, true));
 
                     break;
                 case WT_UPDATE_TOMBSTONE:
                     /*
-                     * When a mixed mode tombstone is getting written to disk, remove any historical
-                     * versions that are greater in the history store for this key.
+                     * When a tombstone without a timestamp is written to disk, remove any
+                     * historical versions that are greater in the history store for this key.
                      */
-                    if (upd_select.mm_tombstone && r->hs_clear_on_tombstone)
+                    if (upd_select.no_ts_tombstone && r->hs_clear_on_tombstone)
```

Comment:
This is not a comment for this PR, but I think there might be an issue here in our current code.
@quchenhao , @kommiharibabu , it seems we are truncating the entries from the history store, if this a non timestamped tombstone with out considering the transaction id. But ideally we should truncate the history store entries only if this tombstone is globally visibile with respect to both timestamp and transaction id.

Reply:
Yes, I agree that is wrong.

Reply:
There is no wrong here. We return the tombstone from rec_upd_select only when the tombstone is globally visible. The flag `no_ts_tombstone` is just an indication of whether the tombstone is a no timestamped or not. Based on that we need to truncate the history store. Let me know If I miss anything.

Reply:
Thanks for checking, I have scheduled an online review tmr to check this issue as the there are two different answers.
@keithbostic, sorry to hijack this PR for a day. As stated earlier, the issue in discussion is not from this PR. Please ignore my comments and proceed with this PR in your usual way. I will track the issue separately.

Reply:
Just to close the loop, we ( myself, Hari & Chenhao) did an online review of the code, and are now satisfied that there is no issue with the logic.